### PR TITLE
golangci-lint  not working correctly

### PR DESCRIPTION
**Pull request by Petrie**
### There's two main changes
- Update golangci-lint version to semver manner in `go.mod` . because dependabot is not able to update non-semver dependency.
- Manual upgrade `golangci-lint` version to fix some linters not working correctly.  Some linters not working correctly on version `v1.45.3-0.20220330010013-d2ccc6d2bbbb`, I don't why, but after upgrade, it works.

### Additionally
As a freshman for metric, i'm not sure this changes is correctly for fix linter on `sdk/metric/aggregator/lastvalue/lastvalue_test.go:79` 

### Related link:
https://github.com/blizzy78/varnamelen/issues/13#issuecomment-1079040934
https://github.com/open-telemetry/opentelemetry-go/pull/2761

### CODE REVIEW

File: sdk/metric/aggregator/lastvalue/lastvalue_test.go [hunk 1]
```diff
@@ -76,7 +76,7 @@ func TestLastValueUpdate(t *testing.T) {
 
 		var last number.Number
 		for i := 0; i < count; i++ {
-			x := profile.Random(rand.Intn(1)*2 - 1)
+			x := profile.Random(rand.Intn(2)*2 - 1)
```
Comment (Aneurysm9) [inline sdk/metric/aggregator/lastvalue/lastvalue_test.go:79]: This doesn't feel right.  What's the lint error being raised here and why is this the approach taken to fix it?

Comment (Petrie) [reply sdk/metric/aggregator/lastvalue/lastvalue_test.go:79]: The lint error is 
```shell
aggregator/lastvalue/lastvalue_test.go:79:24: SA4030: math/rand.Intn(n) generates a random value 0 <= x < n; that is, the generated values don't include n; rand.Intn(1) therefore always returns 0 (staticcheck)
			x := profile.Random(rand.Intn(1)*2 - 1)
			                    ^
```
As my first comment says , As a freshman for metric, I'm not sure this changes is correctly.  pls make some advice directly.

**Comment by Petrie** [comment]
blocked by https://github.com/open-telemetry/opentelemetry-go/issues/2989

**Comment by MrAlias** [comment]
This looks to have been resolve a while ago as far as I can tell. We are currently at version 1.50.1 of golangci-lint: https://github.com/open-telemetry/opentelemetry-go/pull/3404

Closing as this looks superseded. Please reopen if this was in error.

**Pull request closed without merging**